src/sensitivity.jl

```diff
@@ -1,44 +1,48 @@
+    @union_intercepts f type_tuple invoke_type_tuple [kwargs]
 
 Interception strategy based on adding a method to `f` which accepts the union of each of
 the types specified by `type_tuple`. If none of the arguments are `Node`s then the method
-of `f` specified by `invoke_type_tuple` is invoked.
+of `f` specified by `invoke_type_tuple` is invoked. If applicable, keyword arguments
+should be provided as a `NamedTuple` and be added to the generated function's signature.
 """
-macro union_intercepts(f::Symbol, type_tuple::Expr, invoke_type_tuple::Expr)
-    return esc(union_intercepts(f, type_tuple, invoke_type_tuple))
+macro union_intercepts(f::Symbol, type_tuple::Expr, invoke_type_tuple::Expr, kwargs::Expr=:(()))
+    kwargs.head === :tuple || throw(ArgumentError("malformed keyword argument specification"))
+    return esc(union_intercepts(f, type_tuple, invoke_type_tuple; eval(kwargs)...))
 end
 
 """
     union_intercepts(f::Symbol, type_tuple::Expr, invoke_type_tuple::Expr)
 
 The work-horse for `@union_intercepts`.
 """
-function union_intercepts(f::Symbol, type_tuple::Expr, invoke_type_tuple::Expr)
-    call, arg_names = get_union_call(f, type_tuple)
-    body = get_body(f, type_tuple, arg_names, invoke_type_tuple)
-    return Expr(:macrocall, Symbol("@generated"), Expr(:function, call, body))
+function union_intercepts(f::Symbol, type_tuple::Expr, invoke_type_tuple::Expr; kwargs...)
+    call, arg_names = get_union_call(f, type_tuple; kwargs...)
+    body = get_body(f, type_tuple, arg_names, invoke_type_tuple; kwargs...)
+    return Expr(:macrocall, Symbol("@generated"), nothing, Expr(:function, call, body))
 end
 
 """
-    @explicit_intercepts(f::Symbol, type_tuple::Expr, is_node::Expr)
+    @explicit_intercepts(f::Symbol, type_tuple::Expr, is_node::Expr[, kwargs::Expr])
     @explicit_intercepts(f::Symbol, type_tuple::Expr)
 
 Create a collection of methods which intecept the function calls to `f` in which at least
 one argument is a `Node`. Types of arguments are specified by the type tuple expression
 in `type_tuple`. If there are arguments which are not differentiable, they can be
 specified by providing a boolean vector `is_node` which indicates those arguments that are
-differentiable with `true` values and those which are not as `false`.
+differentiable with `true` values and those which are not as `false`. Keyword arguments
+to add to the function signature can be specified in `kwargs`, which must be a `NamedTuple`.
 """
-macro explicit_intercepts(f::SymOrExpr, type_tuple::Expr, is_node::Expr)
-    return esc(explicit_intercepts(f, type_tuple, eval(is_node)))
-end
-macro explicit_intercepts(f::SymOrExpr, type_tuple::Expr)
-    is_node = [true for _ in get_types(get_body(type_tuple))]
-    return esc(explicit_intercepts(f, type_tuple, is_node))
+macro explicit_intercepts(
+    f::SymOrExpr,
+    type_tuple::Expr,
+    is_node::Expr=:([true for _ in $(get_types(get_body(type_tuple)))]),
```

Comment:
Strange code. Can we use `trues` here?

Reply:
It is, though this is effectively reusing code that was already in place, it's just moved around a bit. I think the code requires an array literal, if I'm not mistaken, but even if not, that produces a `BitArray` whereas we need a `Vector{Bool}` to go through most of the machinery here.

Reply:
We can leave this as is.